### ISSUE TITLE
Transfer Exception : Repayment date cannot be on a non working day

### ISSUE THREAD

**Issue by therajanmaurya** [openMF/self-service-app#441]
**Summary:**

I was making normal transfer between savings account and I got the exception from server that
`Repayment date cannot be on a non working day`. I don't think this is a valid exception because self-service is end-user application and user can do any transaction at any time that's the meaning of this app.

```json
{
  "developerMessage": "Request was understood but caused a domain rule violation.",
  "httpStatusCode": "403",
  "defaultUserMessage": "Errors contain reason for domain rule violation.",
  "userMessageGlobalisationCode": "validation.msg.domain.rule.violation",
  "errors": [
    {
      "developerMessage": "Repayment date cannot be on a non working day",
      "defaultUserMessage": "Repayment date cannot be on a non working day",
      "userMessageGlobalisationCode": "error.msg.loan.application.repayment.date.on.non.working.day",
      "parameterName": "id",
      "value": null,
      "args": [
        {
          "value": [
            2017,
            10,
            8
          ]
        }
      ]
    }
  ]
}
```
@nazeer1100126 @edcable What do you think.

**Comment by luckyman20** [comment]
@therajanmaurya I think you're right. This error is valid for only offline transfers. Online transfers can be made on any day. And In this we are making online transactions only.

**Comment by therajanmaurya** [comment]
yes @luckyman000786 you are right, BTW we are building end user bank application so there is nothing like offline transaction, everything will happen in real time.

**Comment by edcable** [comment]
@therajanmaurya let me have @nazeer1100126  or @santoshmath comment as you are correct this validation shouldn't be in place given it's not staff in an office initiating repayment transactions. We'll have to discuss how we can have this validation in place for transactions via web app but not through self-service channels. Can you start a thread on Apache Fineract dev list to publicly discuss this?

**Comment by santoshconflux** [comment]
Discussed with @nazeer1100126 and We too believe that the self service user must be allowed to do transactions on non-working days but it should not allow for office staff (users) if non-working days is configured.

**Comment by santoshconflux** [comment]
@therajanmaurya , Once you create the thread for the discussion as Ed suggested, please share the  link.

**Comment by therajanmaurya** [comment]
Moved on https://mifosforge.jira.com/browse/MM-171
